core/src/test/java/org/apache/calcite/sql/CircularDependencyTest.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to you under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.calcite.sql;
+
+import org.apache.calcite.adapter.enumerable.RexToLixTranslator;
+
+import org.junit.jupiter.api.Test;
+
+import java.util.Map;
+
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.MatcherAssert.assertThat;
+
+/**
+ * Test case for NPE due to circular dependency under certain JVMs.
+ */
+class CircularDependencyTest {
+
+  @Test void test() {
+    Map map = RexToLixTranslator.JAVA_TO_SQL_METHOD_MAP;
+    assertThat(map.isEmpty(), is(false));
```

Comment:
You might probably want to launch the test in isolation (to ensure other tests do not load `RexToLixTranslator`), so you might try adding `--tests org.apache.calcite.sql.CircularDependencyTest` to `./gradlew...` command line in `main.yml`

Reply:
Thanks @vlsi , I'll try that

Reply:
Ah

```
> No tests found for given includes: [**/*Suite*](exclude rules) [org.apache.calcite.sql.CircularDependencyTest](--tests filter)
```

So you need `:core:test --tests ...` rather than `test --tests ...`

Reply:
Yes! Thanks for the hint! I was having the same issue locally, and using `:core:test` solves the issue. Committing the change...

Reply:
Unfortunately, I'm still unable to reproduce the issue. It seems that, in this environment, `InferTypes.FIRST_KNOWN` does not trigger the loading of `SqlCallBinding` (even though it is a parameter of the lambda) , hence breaking the circular dependency produced by `SqlCallBinding#DEFAULT_CALL` initialization.

Reply:
btw, @vlsi , I notice this `macos-latest` in the new `linux-openj9` config in `main.yml`:
```
  linux-openj9:
    if: github.event.action != 'labeled'
    name: 'Linux (OpenJ9 8)'
    runs-on: macos-latest
    ...
```
Shouldn't it be `runs-on: ubuntu-latest`  ? Please confirm and I'll commit the change to re-trigger the tests

Reply:
That looks like a typo. Thanks

Reply:
Thank you very much for your help @vlsi .
Unfortunately, still unable to reproduce the issue. In any case, I will open a Jira and a separate PR to solve the problem.